crates/pathfinder/src/sequencer.rs

```diff
@@ -897,20 +1099,28 @@ mod tests {
 
         #[tokio::test]
         async fn success() {
-            let bytes = client().full_contract(*VALID_CONTRACT_ADDR).await.unwrap();
-            // Fast sanity check
-            // TODO replace with something more meaningful once we figure out the structure to deserialize to
-            assert_eq!(bytes.len(), 53032);
+            use crate::state::contract_hash::json::ContractDefinition;
+            let (_jh, client) = setup([(
+                "/feeder_gateway/get_full_contract?contractAddress=0x6fbd460228d843b7fbef670ff15607bf72e19fa94de21e29811ada167b4ca39",
+                response!("valid_full_contract.json"),
+            )]);
+            let bytes = client.full_contract(*VALID_CONTRACT_ADDR).await.unwrap();
+            serde_json::from_slice::<ContractDefinition>(&bytes).unwrap();
```

Comment:
The test I didn't see when scrolling through.

Reply:
Yep that's the one.